app/controllers/core_induction_programmes/modules_controller.rb

```diff
@@ -7,10 +7,37 @@ class CoreInductionProgrammes::ModulesController < ApplicationController
 
   after_action :verify_authorized
   before_action :authenticate_user!, except: :show
-  before_action :load_course_module
+  before_action :load_course_module, except: %i[new create]
 
   def show; end
 
+  def new
+    authorize CourseModule
+    @course_module = CourseModule.new
+    @course_years = CourseYear.where(core_induction_programme_id: params[:cip_id])
+    @course_modules = CourseModule.where(course_year_id: @course_years.first[:id])
```

Comment:
`CourseModule.where(course_year_id: @course_years.map(&:id))` will give you more than one year :)